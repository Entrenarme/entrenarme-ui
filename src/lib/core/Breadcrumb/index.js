// @flow
import * as React from 'react';
import styled from 'styled-components';

import colors from '../../helpers/colors';
import { regularFont } from '../../helpers/fonts';

const MainContainer = styled.div`
  display: flex;
`;

const LinkContainer = styled.div`
  font-family: ${regularFont};
  font-size: 1rem;
  margin-right: 10px;

  * {
    margin-right: 10px;
    text-decoration: none;
    cursor: ${props => (props.options.disabled ? 'default' : 'pointer')};
    color: ${props =>
      props.options.disabled ? colors.gray.default : colors.gray.placeholder};
    pointer-events: ${props => (props.options.disabled ? 'none' : 'initial')};
  }

  &:hover {
    * {
      color: ${props =>
        props.options.disabled
          ? colors.gray.default
          : colors.secondary.default};
    }
  }

  &::after {
    content: '/';
    color: ${colors.gray.placeholder};
  }

  &:last-child {
    * {
      cursor: default;
      color: ${colors.text};
      margin-right: 0px;
    }
  }

  &:last-child::after {
    content: '';
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (max-width: 767px) {
    font-size: 0.875rem;
    margin-right: 15px;

    * {
      margin-right: 15px;
    }
  }
`;

type Props = {
  children: React.Node,
};

const Breadcrumb = ({ children }: Props) => {
  return (
    <MainContainer>
      {React.Children.map(children, child => (
        <LinkContainer options={{ disabled: child.props.disabled }}>
          {child}
        </LinkContainer>
      ))}
    </MainContainer>
  );
};

export default Breadcrumb;
