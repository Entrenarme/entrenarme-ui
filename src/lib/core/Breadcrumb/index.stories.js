// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';

import Breadcrumb from './index';

storiesOf('Breadcrumb', module).add('with 3 links', () => (
  <Breadcrumb>
    <a>Link 1</a>
    <a>Link 2</a>
    <a>Link 3</a>
  </Breadcrumb>
));

storiesOf('Breadcrumb', module).add('with 5 links', () => (
  <Breadcrumb>
    <a>Link 1</a>
    <a>Link 2</a>
    <a>Link 3</a>
    <a>Link 4</a>
    <a>Link 5</a>
  </Breadcrumb>
));

storiesOf('Breadcrumb', module).add('with disabled links', () => (
  <Breadcrumb>
    <a>Link 1</a>
    <a disabled href="http://www.google.es">Link 2</a>
    <a>Link 3</a>
    <a>Link 4</a>
  </Breadcrumb>
));
