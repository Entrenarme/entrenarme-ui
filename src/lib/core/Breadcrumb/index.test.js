import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import Breadcrumb from './index';

test('renders "Breadcrumb" with no props', () => {
  const { container } = render(
    <Breadcrumb>
      <a>Link 1</a>
      <a>Link 2</a>
      <a>Link 3</a>
    </Breadcrumb>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
