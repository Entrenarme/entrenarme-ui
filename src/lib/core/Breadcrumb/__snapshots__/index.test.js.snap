// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Breadcrumb" with no props 1`] = `
<div
  class="sc-bdVaJa kZAgjJ"
>
  <div
    class="sc-bwzfXH bvfexr"
  >
    <a>
      Link 1
    </a>
  </div>
  <div
    class="sc-bwzfXH bvfexr"
  >
    <a>
      Link 2
    </a>
  </div>
  <div
    class="sc-bwzfXH bvfexr"
  >
    <a>
      Link 3
    </a>
  </div>
</div>
`;
